x-pack/plugins/lens/public/indexpattern_datasource/to_expression.ts

```diff
@@ -164,19 +186,84 @@ function getExpressionForLayer(
       );
     }
 
-    const idMap = esAggEntries.reduce((currentIdMap, [colId, column], index) => {
-      const esAggsId = window.ELASTIC_LENS_DELAY_SECONDS
-        ? `col-${index + (column.isBucketed ? 0 : 1)}-${index}`
-        : `col-${index}-${index}`;
+    const percentileExpressionsByArgs: Record<string, ExpressionAstExpressionBuilder[]> = {};
 
-      return {
-        ...currentIdMap,
-        [esAggsId]: {
-          ...column,
-          id: colId,
-        },
+    aggs.forEach((expressionBuilder) => {
+      const {
+        functions: [fnBuilder],
+      } = expressionBuilder;
+      if (fnBuilder.name === 'aggSinglePercentile') {
+        const field = fnBuilder.getArgument('field')?.[0];
```

Comment:
Need to build key with more than field to make sure only matching percentile aggs are grouped

Reply:
@flash1293 I saw that when I add a timeshift to a single percentile esAgg expression function, it does not get added to the actual aggregation DSL.

Does this mean that an esAggs percentile function with a timeshift can safely be merged together with another that does not have a timeshift as long as the other arguments match?

More broadly, I need to investigate which arguments are compatible for the purposes of optimization.